### PR TITLE
Add "Beta App Description" support for `pilot`

### PR DESCRIPTION
**Pull request by joeydong**
Add support for setting a "Beta App Description" when uploading builds using `pilot`

This is required when uploading builds for the first time. I've also seen iTunes Connect "forget" the beta app description between versions so it needs to be set again.

This should solve https://github.com/fastlane/fastlane/issues/6351 and https://github.com/fastlane/fastlane/issues/6117

### CODE REVIEW

**Review by KrauseFx — APPROVED** [review]

**Comment by KrauseFx** [comment]
This looks great, thanks 👍 I wasn't aware that _spaceship_ can already do that, we just didn't use it. 

Have you tried this locally too?

**Comment by matteo-harpoon** [comment]
I'm trying to send a Testflight build to both internal and external uses and the message I get right now is the following
```
[!] The request could not be completed because:
	[English]: What to Test is required.
[English]: Beta App Description is required.
[English]: Feedback emails are required.
[English]: Feedback emails must be in the format: contact@example.com.
[English]: The marketing URL you entered is invalid. Make sure you entered it correctly and try again.
[English]: Marketing URLs are required. The phone number is missing. The email is required. The email must be in the format: contact@example.com. Enter a demo account user name, or deselect "A demo account is required to use all the features of the app." Enter demo account password, or deselect "A demo account is required to use all the features of the app." What to Test is required. Beta App Description is required. Feedback emails are required. Feedback emails must be in the format: contact@example.com. The marketing URL you entered is invalid. Make sure you entered it correctly and try again. Marketing URLs are required. The email is required. The email must be in the format: contact@example.com. The phone number is missing. Enter a demo account user name, or deselect "A demo account is required to use all the features of the app." Enter demo account password, or deselect "A demo account is required to use all the features of the app." Validation errors
```

Now except for the `What to Test` and `Beta App Description` (which you are currently working on), I think it should be also possible to send via metadata the following variables:
- `Test Information > en-US > Feedback Email`
- `Test Information > en-US > Marketing URL`
- `Test Information > en-US > Privacy Policy URL`
- `Test Information > en-US > License Agreement` (optional)

**Comment by joeydong** [comment]
@matteo-harpoon Yeah there's definitely more fields we should add to fully support the first-run experience. I'm just pushing the "Beta App Description" one for now because iTunes Connect would ask our app to set it every version for some reason (some bug on their end).

**Comment by joeydong** [comment]
@KrauseFx yup I've been sitting on this change for a while. I haven't tried it with the `.positive` syntax but I'll have a chance to test it out for our build early next week.

File: pilot/lib/pilot/build_manager.rb [hunk 1]
```diff
@@ -63,10 +63,10 @@ def distribute(options, build = nil)
         UI.message("Distributing build #{build.train_version}(#{build.build_version}) from #{build.testing_status} -> External")
       end
 
-      # First, set the changelog (if necessary)
-      if options[:changelog].to_s.length > 0
-        build.update_build_information!(whats_new: options[:changelog])
-        UI.success "Successfully set the changelog for build"
+      # First, set the changelog and/or description if necessary
+      if options[:changelog].to_s.length.positive or options[:beta_app_description].to_s.length.positive
```
Comment (KrauseFx) [inline pilot/lib/pilot/build_manager.rb]: `positive` is is not an available method, this might be Rails specific?

**Review by KrauseFx — CHANGES_REQUESTED** [review]

File: pilot/lib/pilot/build_manager.rb (hunk 1 above)
Comment (joeydong) [inline pilot/lib/pilot/build_manager.rb]: Hmm okay. I just put it back to `> 0`. The houndci-bot might need some silencing then.

Comment (joeydong) [inline pilot/lib/pilot/build_manager.rb]: Maybe I was missing the `?` in `positive?`.

**Comment by joeydong** [comment]
I just used the tool this morning with the new `beta_app_description` flag and it worked as expected.

**Review by KrauseFx — APPROVED** [review]

**Comment by KrauseFx** [comment]
Looks great, thanks for your contribution and updating the PR 👍

**Comment by KrauseFx** [comment]
This will be in the next release

**Review by ryzn0518 — COMMENTED** [review]
@joeydong  Can you tell me how to add `beta_app_description` when `pilot`,as I had the question
`Beta App Description is required. [English]: Beta App Description is required`
 As I enter the command `pilot -h`,I had not see any like`--beta_app_descripiton`,how can I add beta_app_descripiton in the command?

File: pilot/lib/pilot/build_manager.rb [hunk 2]
```diff
@@ -63,10 +63,10 @@ def distribute(options, build = nil)
...
 
-      # First, set the changelog (if necessary)
-      if options[:changelog].to_s.length > 0
-        build.update_build_information!(whats_new: options[:changelog])
-        UI.success "Successfully set the changelog for build"
+      # First, set the changelog and/or description if necessary
+      if options[:changelog].to_s.length > 0 or options[:beta_app_description].to_s.length > 0
+        build.update_build_information!(whats_new: options[:changelog], description: options[:beta_app_description])
+        UI.success "Successfully set the changelog and/or description for build"
       end
```
Comment (joeydong) [inline pilot/lib/pilot/build_manager.rb]: Looks like a new release was just tagged 20 hours ago. Did you update the `pilot` gem?

Comment (joeydong) [inline pilot/lib/pilot/build_manager.rb]: Oh we're waiting for the spec to be updated at https://rubygems.org/gems/pilot

**Pull request merged**